indexing-service/src/main/java/io/druid/indexing/overlord/autoscaling/ScalingStats.java

```diff
@@ -65,6 +65,15 @@ public ScalingStats(int capacity)
     }
   }
 
+  public ScalingStats(ScalingStats stats1, ScalingStats stats2)
+  {
+    this(stats1.recentEvents.size() + stats2.recentEvents.size());
+    synchronized (lock) {
```

Comment:
Redundant, we cannot create a single object with 2 threads.
Also, careful, we need somehow to get sizes after a toList() call. Here would be better a constructor like
`new ScalingStats(stats1.toList(), stats2.toList())`